src/routes/group.js

```diff
@@ -0,0 +1,22 @@
+import express from 'express';
+import Group from '../models/group';
+
+const groups = express.Router();
+
+// add a new group
+groups.post('/', async (req, res) => {
+  const newGroup = new Group(
+    {
+      inviteCode: req.body.inviteCode,
```

Comment:
We generate the invitecode, it is not provided by the request. It should be generated using the `_id` generated once we save this group. So the workflow is: save the group without the inviteCode -> get the `_id` from the saved document -> generate the invite code based on the hashing of that `_id` -> update the newly saved document with this generated inviteCode.

Reply:
We may want to do invite-related stuff in a separate PR